src/Compilers/CSharp/Portable/BoundTree/DecisionTreeBuilder.cs

```diff
@@ -108,14 +105,11 @@ private DecisionTree AddByValue(DecisionTree.Guarded guarded, BoundConstantPatte
         {
             if (guarded.Default != null)
             {
-                if (guarded.Default.MatchIsComplete)
-                {
-                    return null;
-                }
+                Debug.Assert(!guarded.Default.MatchIsComplete); // otherwise we would have given a subsumption error
             }
             else
             {
-                guarded.Default = new DecisionTree.ByValue(guarded.Expression, guarded.Type, null);
+                guarded.Default = new DecisionTree.ByType(guarded.Expression, guarded.Type, null);
```

Comment:
Perhaps, it is still not clear why it should be initialized in this particular way. I assume the previous code was incorrect. Why this is correct in general, for all possible scenarios?

---
In reply to: [100598334](https://github.com/dotnet/roslyn/pull/16928#discussion_r100598334) [](ancestors = 100598334,100474813)